src/mixpanel-core.js

```diff
@@ -541,10 +557,21 @@ MixpanelLib.prototype.disable = function(events) {
  *
  * @param {String} event_name The name of the event. This can be anything the user does - 'Button Click', 'Sign Up', 'Item Purchased', etc.
  * @param {Object} [properties] A set of properties to include with the event you're sending. These describe the user who did the event or details about the event itself.
+ * @param {Object} [options] Optional configuration for this track request.
+ * @param {String} [options.transport] Transport method for network request ('xhr' or 'sendBeacon').
  * @param {Function} [callback] If provided, the callback function will be called after tracking the event.
  */
-MixpanelLib.prototype.track = addOptOutCheckMixpanelLib(function(event_name, properties, callback) {
-    if (typeof(callback) !== 'function') {
+MixpanelLib.prototype.track = addOptOutCheckMixpanelLib(function(event_name, properties, options, callback) {
+    if (!callback && _.isFunction(options)) {
+        callback = options;
+        options = null;
+    }
+    options = options || {};
+    var transport = options['transport'];
+    if (transport) {
+        options.transport = transport;
+    }
```

Comment:
I'm confused by this, missing how it modifies `options`. Can you add a clarifying comment if it's intentional?

Reply:
Will do - it's because `track()` is a public API and so "transport" shouldn't be minified, whereas `_send_request()` is internal and so its `options.transport` can be minified however closure compiler wants.

Reply:
that makes a strange amount of sense, thanks for the explanation